documentation/src/main/asciidoc/cassandraclient.adoc

```diff
@@ -0,0 +1,301 @@
+}
+----
+We are using ObjectMapper, due to that fact the class is annotated with an `@Entity`.
+Also, the Fruit `id` is a `@PartitionKey` and `name` is a `@ClusteringColumn` so we are using the corresponding annotations
+from the ObjectMapper. It will allow the Mapper to generate proper CQL queries underneath.
+
+One important thing to note is that having a default constructor is required by the Object Mapper.
+
+To leverage the Mapper logic in this app we need to create a DAO:
+[source,java]
+----
+@Dao
+public interface FruitDao {
+  @Update
+  void update(Fruit fruit);
+
+  @Select
+  PagingIterable<Fruit> findById(String id);
+}
+----
+
+This class exposes operations that will be used in the REST service.
+
+Finally, the Mapper itself:
+[source,java]
+----
+@Mapper
+public interface FruitMapper {
+  @DaoFactory
+  FruitDao fruitDao(@DaoKeyspace CqlIdentifier keyspace);
+}
+----
+It is constructing the `FruitDao`.
+
+Now create a `com.datastax.oss.quarkus.FruitService` that will be the business layer of our application and store/load the fruits from the Cassandra database.
+[source,java]
+----
+@ApplicationScoped
+public class FruitService {
+  private final FruitDao dao;
+
+  @Inject
+  public FruitService(CqlSession session) {
+    dao = new FruitMapperBuilder(session).build().fruitDao(CqlIdentifier.fromCql("k1"));
+  }
+
+  public void save(Fruit fruit) {
+    dao.update(fruit);
+  }
+
+  public List<Fruit> get(String id) {
+    return dao.findById(id).all();
+  }
+}
+----
+Please note the construction of dao. It is using `CqlSession` injected automatically by the cassandra-quarkus extension.
+
+The last missing piece is the REST API that will expose GET and POST methods:
+[source,java]
+----
+@Path("/fruits")
+@Produces(MediaType.APPLICATION_JSON)
+@Consumes(MediaType.APPLICATION_JSON)
+public class FruitResource {
+
+  private static final String STORE_NAME = "acme";
+  @Inject FruitService fruitService;
+
+  @GET
+  public List<Fruit> list() {
+    return fruitService.get(STORE_NAME);
+  }
+
+  @POST
+  public void add(FruitDto fruit) {
+    fruitService.save(covertFromDto(fruit));
+  }
+
+  private Fruit covertFromDto(FruitDto fruitDto) {
+    return new Fruit(fruitDto.getName(), fruitDto.getDescription(), STORE_NAME);
+  }
+}
+----
+
+The list and add operations are executed for the `STORE_NAME` id. This is a partition key of the data model.
+We can easily retrieve all rows from cassandra using the partition key. They will be sorted by the clustering key.
+The `FruitResource` is using the `FruitService` that encapsulates the data access logic.
+
+When creating the REST API we should not share the same entity object between REST API and data access layers.
+They should not be coupled to allow evolving API independent of the storage layer.
+This is the reason why the API is using the `FruitDto`.
+This class will be used by the quarkus to convert JSON to java object for client request and java object to JSON for the response.
```

Comment:
Nit: s/used by the quarkus/used by Quarkus/

This translation is actually provided by REST Easy (https://resteasy.github.io/) so we might just want to say that here